Mathlib/Topology/Connected/PathConnected.lean

```diff
@@ -1215,4 +1227,71 @@ theorem IsOpen.isConnected_iff_isPathConnected {U : Set X} (U_op : IsOpen U) :
+lemma locPathConnectedSpace_iff_isOpen_pathComponentIn {X : Type*} [TopologicalSpace X] :
+    LocPathConnectedSpace X ↔ ∀ (x : X) (u : Set X), IsOpen u → IsOpen (pathComponentIn x u) :=
+  ⟨fun _ _ _ hu ↦ hu.pathComponentIn _, fun h ↦ ⟨fun x ↦ ⟨fun s ↦ by
+    refine ⟨fun hs ↦ ?_, fun ⟨_, ht⟩ ↦ Filter.mem_of_superset ht.1.1 ht.2⟩
+    let ⟨u, hu⟩ := mem_nhds_iff.mp hs
+    exact ⟨pathComponentIn x u, ⟨(h x u hu.2.1).mem_nhds (mem_pathComponentIn_self hu.2.2),
+      isPathConnected_pathComponentIn hu.2.2⟩, pathComponentIn_subset.trans hu.1⟩⟩⟩⟩
+
+/-- A space is locally path-connected iff all path components of open subsets are neighbourhoods. -/
+lemma locPathConnectedSpace_iff_pathComponentIn_mem_nhds {X : Type*} [TopologicalSpace X] :
+    LocPathConnectedSpace X ↔
+    ∀ x : X, ∀ u : Set X, IsOpen u → x ∈ u → pathComponentIn x u ∈ nhds x := by
+  rw [locPathConnectedSpace_iff_isOpen_pathComponentIn]
+  simp_rw [forall_comm (β := Set X), ← imp_forall_iff]
+  refine forall_congr' fun u ↦ imp_congr_right fun _ ↦ ?_
+  exact ⟨fun h x hxu ↦ (h x).mem_nhds (mem_pathComponentIn_self hxu),
+    fun h x ↦ isOpen_iff_mem_nhds.mpr fun y hy ↦
+      pathComponentIn_congr hy ▸ h y <| pathComponentIn_subset hy⟩
+
+/-- Any topology coinduced by a locally path-connected topology is locally path-connected. -/
+lemma LocPathConnectedSpace.coinduced {Y : Type*} (f : X → Y) :
+    @LocPathConnectedSpace Y (.coinduced f ‹_›) := by
+  let _ := TopologicalSpace.coinduced f ‹_›; have hf : Continuous f := continuous_coinduced_rng
+  refine locPathConnectedSpace_iff_isOpen_pathComponentIn.mpr fun y u hu ↦
+    isOpen_coinduced.mpr <| isOpen_iff_mem_nhds.mpr fun x hx ↦ ?_
+  have hx' := preimage_mono pathComponentIn_subset hx
+  refine mem_nhds_iff.mpr ⟨pathComponentIn x (f ⁻¹' u), ?_,
+    (hu.preimage hf).pathComponentIn _, mem_pathComponentIn_self hx'⟩
+  rw [← image_subset_iff, ← pathComponentIn_congr hx]
+  exact ((isPathConnected_pathComponentIn hx').image hf).subset_pathComponentIn
+    ⟨x, mem_pathComponentIn_self hx', rfl⟩ <|
+    (image_mono pathComponentIn_subset).trans <| u.image_preimage_subset f
+
+/-- Quotients of locally path-connected spaces are locally path-connected. -/
+lemma Topology.IsQuotientMap.locPathConnectedSpace {f : X → Y} (h : IsQuotientMap f) :
+    LocPathConnectedSpace Y :=
+  h.2 ▸ LocPathConnectedSpace.coinduced f
+
+/-- Quotients of locally path-connected spaces are locally path-connected. -/
+instance Quot.locPathConnectedSpace {r : X → X → Prop} : LocPathConnectedSpace (Quot r) :=
+  isQuotientMap_quot_mk.locPathConnectedSpace
+
+/-- Quotients of locally path-connected spaces are locally path-connected. -/
+instance Quotient.locPathConnectedSpace {s : Setoid X} : LocPathConnectedSpace (Quotient s) :=
+  isQuotientMap_quotient_mk'.locPathConnectedSpace
+
+/-- Disjoint unions of locally path-connected spaces are locally path-connected. -/
+instance Sigma.locPathConnectedSpace {X : ι → Type*}
```

Comment:
Please also add an instance for `Sum`. BTW, you may want to generalize the instance `[NormedSpace Real V] : LocPathConnectedSpace V` to `LocallyConvexSpace`.

Reply:
I've added the instance for `Sum` now - its proof seems currently a bit redundant, but is the cleanest of several variants I tried.

A `LocPathConnectedSpace`-instance for locally convex spaces is also in #17142 already which is only blocked by this PR, so I don't think there's much point in moving it here.